Arrangement_on_surface_2/include/CGAL/Arr_segment_traits_2.h

```diff
@@ -82,17 +82,13 @@ class Arr_segment_traits_2 : public Kernel_ {
     typedef typename Kernel::Point_2               Point_2;
 
   protected:
-    mutable Line_2 m_l;       // the line that supports the segment.
-    Point_2 m_ps;             // the source point of the segment.
-    Point_2 m_pt;             // the target point of the segment.
-    bool m_is_directed_right; // is (lexicographically) directed left to right.
-    mutable enum              // is this a vertical segment.
-    {
-      UNKNOWN,                  // If the line is not computed
-      REGULAR,                  // If the line is computed and not vertical
-      VERTICAL                  // If the line is computed and vertical
-    } m_is_vert;
-    bool m_is_degen;          // is the segment degenerate (a single point).
```

Comment:
The underlying type of an enum is implementation defined. However, since C++11, one can specify it:
```c++
enum Is_vert : char { /* ... */ ];
```

Actually, the size of a `bool` is also implementation-defined. The version with the `enum` might be smaller.

And be also careful of alignement: the `class` can be smaller if the two or four `bool` are packed together.

Reply:
The class being smaller with two of four `bool` packed together is pretty much what Efi analyzed: in the case of this class, there was already 2 `bool`, so adding a third one does not change the size.

I did not know about the specification of the enum type, it's very interesting and good to know! In the case of this PR, as the size won't be smaller, I think using 2 booleans remains more readable (and the change is less invasive than replacing the existing boolean with an enum).

Reply:
No problem. I use feedbacks on issues to "trickle down" things I know on C++. In the context of this issue, I understand why using the booleans can be more useful or efficient.